src/dmd/expression.d

```diff
@@ -1769,9 +1764,8 @@ extern (C++) abstract class Expression : RootObject
         {
             a = new Expressions();
             a.setDim(exps.dim);
-            for (size_t i = 0; i < a.dim; i++)
+            foreach (i, e; *exps)
             {
-                Expression e = (*exps)[i];
                 (*a)[i] = e ? e.syntaxCopy() : null;
```

Comment:
Would be could to have `lockstep` / `zip` here, s.t. we can drop the iteration index in favor of the second range to be iterated.